addons/l10n_vn/migrations/16.0.2.0.2/post-migration.py

```diff
@@ -0,0 +1,28 @@
+from odoo import api, SUPERUSER_ID
+from odoo.osv import expression
+
+FIXED_ACCOUNTS_TYPE = {
+    'expense_direct_cost': ['6271', '6272', '6273', '6274', '6277', '6278'],
+}
+
+
+def _fix_accounts_type(env):
+    for correct_account_type, accounts_code in FIXED_ACCOUNTS_TYPE.items():
+        domains_per_company = []
+        for company in env['res.company'].with_context(active_test=False).search([('chart_template', '=', 'vn')]):
+            domain = expression.AND([
+                [('company_id', '=', company.id), ('account_type', '!=', correct_account_type)],
+                expression.OR([
+                    [('code', 'like', f'{code}%')] for code in accounts_code
+                ])
+            ])
+            domains_per_company.append(domain)
+        accounts = env['account.account'].search(expression.OR(domains_per_company))
+        accounts.write({
+            'account_type': correct_account_type
+        })
```

Comment:
Why are we creating the domain per company? How about something like this instead? (plz also test it first to make sure it works)

```py
def _fix_accounts_type(env):
    vn_companies = env['res.company'].with_context(active_test=False).search([('chart_template', '=', 'vn')])

    for correct_account_type, account_codes in FIXED_TYPE_CODES_MAP.items():
        accounts = env['account.account'].search([
            ('company_id', 'in', vn_companies.ids),
            ('account_type', '!=', correct_account_type),
            expression.OR([
                [('code', '=like', f'{code}%')]
                for code in account_codes
            ]),
        ])
        accounts.write({'account_type': correct_account_type})

Reply:
thanks, I tested everything ok.